src/runtime/threads/policies/scheduler_base.cpp

```diff
@@ -51,6 +52,18 @@ namespace hpx { namespace threads { namespace policies
       , parent_pool_(nullptr)
       , background_thread_count_(0)
     {
+        // if there is a cfg setting for the default_scheduler_mode, use that
+        // instead of the built-in default
+        std::string default_scheduler_mode = hpx::get_config_entry(
+            "hpx.default_scheduler_mode", std::string());
+
+        if (!default_scheduler_mode.empty())
+        {
+            mode = scheduler_mode(hpx::util::safe_lexical_cast<std::size_t>(
+                default_scheduler_mode));
+            HPX_ASSERT((mode & ~scheduler_mode::all_flags) == 0);
```

Comment:
A message would probably be good here saying that the config entry contains illegal scheduler options.

Reply:
Yah, I was thinking of throwing an exception, I might just do that instead.

Reply:
Yeah, that works. I was thinking of `HPX_ASSERT_MSG` but either is fine with me.